wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,462 @@
+	- Event Damage is set to .actualAmount, not .amount.
+	- Damage instance can not be modified from inside listeners.
+
+	There will be output to debug log stream if anything would go wrong,
+	for example if system will fail to pass/catch Damage instance in between
+	damage instance creation and it's handling (.deal() and .handle()).
+	
+	Examples:
+	
+	// Leech implementation #1
+	function onLeechEffect1()
+		// Using wrapper from the package, event data is retrieved easily.
+		let d = GetTriggerUnit().damageTarget(GetSpellTargetUnit(), LEECH_DAMAGE, ATTACK_TYPE_NORMAL, 0)
+
+		d.source.addHP(d.getActual()) // Heal is capped at target's hp
+		d.source.addEffect(Abilities.vampiricAuraTarget, "origin").destr()
+
+		d.releaseInstance() // Don't forget to free the instance
+
+	init
+		registerSpellEffectEvent(LEECH_ID, function onLeechEffect1)
+
+	// Leech implementation #2
+	function onLeechEffect2()
+		GetTriggerUnit().damageTarget(GetSpellTargetUnit(), LEECH_DAMAGE, ATTACK_TYPE_NORMAL, LEECH_ID)
+
+	init
+		registerSpellEffectEvent(LEECH_ID, function onLeechEffect2)
+		addDamageListener() d ->
+			if d.data == LEECH_ID
+				d.source.addHP(d.getActual())
+				d.source.addEffect(Abilities.vampiricAuraTarget, "origin").destr()
+
+	// Simple damage resistance buff
+	import UniqueDamageListening
+
+	function onBuffEffect()
+		let target = GetSpellTargetUnit()
+		let effect = target.addEffect(Abilities.voodooAuraTarget, "origin")
+		let cb = addTargetDamageModifier() d ->
+			d.amount -= d.initialAmount * .5
+		doAfter(BUFF_DURATION) ->
+			removeTargetDamageModifier(cb)
+			effect.destr()
+
+	init
+		registerSpellEffectEvent(BUFF_ID, function onLeechEffect2)
+*/
+
+// Configuration
+
+/** If Damage instance wasn't created by one of DamageModify package's functions
+	(or `new Damage(...)`), then Damage.data would be set to this value */
+@configurable public constant DAMAGE_NONTRIGGERED_DATA = -1
+
+/** Maximum amount of damage events that can happen in a single event */
+@configurable public constant RECURSION_TRESHOLD = 32
+
+/** Describes default way to create Damage instance if damage event doesn't
+	correspond to any Damage instance. */
+@configurable function getDamageFromEvent() returns Damage
+	return new Damage(GetEventDamageSource(), GetTriggerUnit(), GetEventDamage(), false, false, getDamageType() == DamageType.SPELL ? ATTACK_TYPE_NORMAL : ATTACK_TYPE_CHAOS, null, null, DAMAGE_NONTRIGGERED_DATA, false)
+
+// Wrappers for unit.damageTarget with data attachment
+
+/** Deals triggered damage from unit to target. Additional data can be specified.
+
+	Is a helper for `new Damage(*args*, false)..modify()..deal()`, where last
+	arg specifies whether the instance should not be destroyed during the event.
+	If you need instance after it has been dealt, create it manually. */
+public function unit.damageTarget(unit target, real amount, bool attack, bool ranged, attacktype attackType, damagetype damageType, weapontype weaponType, int data)
```

Comment:
these names should be differentiated so you don't mix them up with the normal damageTarget.

Reply:
1) It was intended, although I can't support that intention with any arguments.
2) Could you suggest a different name?

Reply:
I think something like `dealCodeDamage` from DamageType is pretty on point.